src/service/xhr-impl.js

```diff
@@ -407,13 +427,11 @@ function isRetriable(status) {
  */
 export function assertSuccess(response) {
   return new Promise((resolve, reject) => {
-    if (response.status < 200 || response.status >= 300) {
-      /** @const {!Error} */
-      const err = user().createError(`HTTP error ${response.status}`);
-      err.response = response;
-      if (isRetriable(response.status)) {
-        err.retriable = true;
-      }
+    const status = response.status;
+    if (status < 200 || status >= 300) {
+      const retriable = isRetriable(status);
+      const err = user().createCustomError(
```

Comment:
well, `createError` don't report but whoever is catching `.fetch()` error may report it. (or we might be reporting all `unhandledrejection` error, @jridgewell do we report `unhandledrejection`?)

 You are right that `response` does got get reported right now because it is just a field on FetchError, but do we need it? I like dump error objects that one can't do anything but log them instead of error objects that can be inspected and used to switch code path. (e.g. not a fan of `.catch(err => if( err.response.code == 500) {...} )`